debug/src/component-stack.js

```diff
@@ -87,12 +87,12 @@ export function getOwnerStack(vnode) {
 		const source = owner.__source;
 		if (source) {
 			acc += ` (at ${source.fileName}:${source.lineNumber})`;
-		} else if (!hasBabelPlugin) {
-			hasBabelPlugin = true;
+		} else if (showJsxSourcePluginWarning) {
 			console.warn(
 				'Add @babel/plugin-transform-react-jsx-source to get a more detailed component stack. Note that you should not add it to production builds of your App for bundle size reasons.'
 			);
 		}
+		showJsxSourcePluginWarning = false;
```

Comment:
The idea being that at this point we've either seen `.__source` or printed the warning, and in either case, we're now covered. `hasBabelPlugin` previously would only ever be set to `true` once the warning was shown.